src/CitMovie.Data/Repositories/GenreRepository.cs

```diff
@@ -0,0 +1,31 @@
+using Microsoft.EntityFrameworkCore;
+
+namespace CitMovie.Data.Repositories;
+
+public class GenreRepository : IGenreRepository
+{
+    private readonly DataContext _context;
+
+    public GenreRepository(DataContext context)
+    {
+        _context = context;
+    }
+    public async Task<IList<Genre>> GetAllGenresAsync(int page, int pageSize)
+    {
+        return await _context.Genres

```

Comment:
Adding .AsNoTracking() before Skip would improve performance. AsNoTracking tells Entity Framework that we don’t intend to update these entities, so it avoids tracking them in the context.

```suggestion
        return await _context.Genres
        .AsNoTracking()
```

Reply:
Exactly!
            .AsNoTracking() // Should be placed before `skip()` and `Take()`
            .Skip(page * pageSize)
            .Take(pageSize)
            
Basically what `AsNoTracking()` does is it prevents Entity Framework from tracking changes to the Country entities in the returned results. It removes a lot of overhead and makes everything run smoother.